lib/components/steps/Home.tsx

```diff
@@ -1,17 +1,38 @@
+import React from 'react';
 import { Box, Button, Typography, Chip, TableContainer, Paper } from '@mui/material';
 import { StyledTable } from '../common/StyledTable.tsx';
 import { CircleChipDefault, CircleChipSuccess } from '../../icons/index.tsx';
 import { vars } from '../../theme/variables.ts';
 import { STEPS } from "../../models.ts";
 import { useCdeContext } from "../../CdeContext.ts";
+import { isRowMapped } from '../../helpers/getters.ts';
 
 const { primary600, gray600, drodownDetailBg, gray200 } = vars;
 
 function Home() {
     const {
         setStep,
-        datasetSample
+        datasetSample,
+        name,
+        datasetMapping,
+        headerIndexes,
+        getSuggestions
     } = useCdeContext();
+    
+    const [numberOfMapped, setNumberOfMapped] = React.useState(0);
+    const suggestionsMapping = getSuggestions();
```

Comment:
question: @Aiga115 why is this done using a getter function and is the datasetMapping not done using a getter function

Reply:
I used getSuggestions function as it was mentioned in a description of a task -> https://metacell.atlassian.net/jira/software/c/projects/CDE/boards/44?selectedIssue=CDE-60
if you follow a link in a description of a task it will get you to function's reference 
But is there a getter function to get dataset mapping? I don't think there is @zsinnema

Reply:
I will look into this one @Aiga115

Reply:
@zsinnema please have a look at the changes done in commit [1d6e643](https://github.com/MetaCell/cde-mapper/pull/26/commits/1d6e643b98e6f1a24c1fa395c8ad714a310ed84e). Based on our conversation, I decided to split the cde context into 3 different contexts; each with a different purpose:

Data Context: Holds the application's core data;
Service Context: Encapsulates functions and operations that act upon the data;
UI Context: Manages UI-related state and interaction;
For the components that we will create, if they need the data exactly as it is stored, I believe it's fine to use the data context; However, if the component needs to perform some transformation or do some calculations based on the data stored it should use the service context.